src/main/java/pp/springdata/todo/TodoController.java

```diff
@@ -0,0 +1,89 @@
+package pp.springdata.todo;
+
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Controller
+public class TodoController {
+
+    private final TodoRepository todoRepository;
+
+    public TodoController(TodoRepository todoRepository) {
+        this.todoRepository = todoRepository;
+    }
+
+    @GetMapping("/")
+    public String home(Model model, @RequestParam(required = false) Category category,
+                       @RequestParam(required = false, defaultValue = "false") Boolean archive) {
+        List<Todo> todoList;
+        if (category != null) {
+            todoList = todoRepository.findByCategoryOrderByDeadline(category);
+        } else {
+            todoList = todoRepository.findAll(Sort.by("deadline"));
+        }
+
+        model.addAttribute("todoList", todoList.stream()
+                .filter(todo -> todo.getDone().equals(archive))
```

Comment:
A to mogłaś przenieść na bazę danych. Generalnie staramy się jak najmniej danych wyciągać z bazy. W Twoim przypadku i tak zawsze je filtrujesz po tym czy są skończone czy nie to możesz to wrzucić do metody Spring Data.